rockcraft.yaml

```diff
@@ -0,0 +1,183 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+---
+
+name: charmed-ranger-rock
+base: ubuntu@22.04
+version: 2.4.0-22.04-edge
+summary: Charmed ranger ROCK OCI
+description: |
+  Apache Ranger™ is a framework to enable, monitor and
+  manage comprehensive data security across the Hadoop platform.
+license: Apache-2.0
+
+platforms:
+  amd64:
+
+# Please refer to
+# https://discourse.ubuntu.com/t/unifying-user-identity-across-snaps-and-rocks/36469
+# for more information about shared user.
+run_user: _daemon_
+
+services:
+  ranger-admin:
+    override: replace
+    summary: "ranger admin service"
+    startup: disabled
+    command: "/home/ranger/scripts/ranger-admin-entrypoint.sh"
+    environment:
+      JAVA_OPTS: "-Duser.timezone=UTC0"
+  ranger-usersync:
+    override: replace
+    summary: "ranger usersync service"
+    startup: disabled
+    command: "/home/ranger/scripts/ranger-usersync-entrypoint.sh"
+
+parts:
+  base:
+    plugin: dump
+    source: "https://downloads.apache.org/ranger/2.4.0/apache-ranger-2.4.0.tar.gz" # yamllint disable-line
+    source-checksum: "sha512/5bc934416e7239e2b74843399389a69e8c129322ff0a67b2e7a541a4dd2171357b10925ea29861c40e7590a6523494d46343fdda38049d9a66e48d80eed11a4b" # yamllint disable-line
+    build-packages:
+      - build-essential
+      - curl
+      - maven
+      - git
+      - openjdk-11-jdk-headless
+    build-environment:
+      - JAVA_HOME: /usr/lib/jvm/java-11-openjdk-amd64
+      - RANGER_DIST: /home/ranger/dist
+    override-build: |
+      craftctl default
+
+      # Build Ranger
+      mvn clean compile package install \
+        -DskipTests=true \
+        -Dmaven.test.skip=true \
+        -Drat.skip=true \
+        -Dpmd.skip=true \
+        -Dfindbugs.skip=true \
+        -Dspotbugs.skip=true \
+        -Dcheckstyle.skip=true \
+        -Dmaven.wagon.http.ssl.insecure=true \
+        -Dmaven.wagon.http.ssl.allowall=true \
+        -Dmaven.wagon.http.ssl.ignore.validity.dates=true
+
+      # Make dist directory
+      mkdir -p ${CRAFT_PART_INSTALL}${RANGER_DIST}
+      mv target ${CRAFT_PART_INSTALL}${RANGER_DIST}/target
+    stage:
+      - home/ranger/dist/target/ranger-2.4.0-admin.tar.gz
```

Comment:
I see that these are used by the `admin` part, but do they need to be included in the final rock too?

Reply:
No they don't, if I don't want them to be in the final rock what would be the approach here. To remove them after use in the `admin` section? `rm -f ${CRAFT_STAGE}${RANGER_DIST}/ranger-${VERSION}-admin.tar.gz`. Or should I not be staging them this way?

Reply:
You can add this:

```
    prime:
      - "-*"
```

Which looks odd due to the yaml syntax but reads "exclude everything". So the files will be available during the stage step and then removed on prime.